thonny/code.py

```diff
@@ -241,7 +245,7 @@ def save_file(self, ask_filename=False):
             self._filename = new_filename
             get_workbench().event_generate("SaveAs", editor=self, filename=new_filename)
 
-        content = self._code_view.get_content_as_bytes()
+        content = self._code_view.get_content_as_bytes(self._newlines == '\r\n')
```

Comment:
Did you know that `filenames` attribute can be a tuple? (https://stackoverflow.com/a/29832736/261181)

Anyway, I think it is good as it is now, because it makes sense to use platform format if the original file contained more than one type of linebreak.